convert.c

```diff
@@ -256,6 +257,135 @@ static int will_convert_lf_to_crlf(size_t len, struct text_stat *stats,
 
 }
 
+#ifdef NO_ICONV
+#ifndef _ICONV_T
+/* The type is just a placeholder and not actually used. */
+typedef void* iconv_t;
+#endif
+#endif
+
+static struct encoding {
+	const char *name;
+	iconv_t to_git;       /* conversion to Git's canonical encoding (UTF-8) */
+	iconv_t to_worktree;  /* conversion to user-defined encoding */
+	struct encoding *next;
+} *encoding, **encoding_tail;
+static const char *default_encoding = "utf-8";
+static iconv_t invalid_conversion = (iconv_t)-1;
+
+static int encode_to_git(const char *path, const char *src, size_t src_len,
```

Comment:
I wonder whether this should maybe be called `encode_for_git` (as well as `encode_for_worktree`). `encode_to_git` could imply that Git has or is a custom encoding format, while this really is just UTF-8.

Reply:
maybe. but this is kind of in line with the other functions I think